front/src/applications/stdcmV2/components/StdcmOperationalPoint.tsx

```diff
@@ -0,0 +1,174 @@
+import SelectSNCF from 'common/BootstrapSNCF/SelectSNCF';
+import useSearchOperationalPoint, {
+  MAIN_OP_CH_CODES,
+} from 'common/Map/Search/useSearchOperationalPoint';
+import type { PathStep } from 'reducers/osrdconf/types';
+import { useAppDispatch } from 'store';
+
+import StdcmSuggestions from './StdcmSuggestions';
+
+type UpdatePointActions =
+  | ActionCreatorWithPayload<PathStep | null, 'stdcmConf/updateOriginV2'>
+  | ActionCreatorWithPayload<PathStep | null, 'stdcmConf/updateDestinationV2'>;
+
+type StdcmOperationalPointProps = {
+  updatePoint: UpdatePointActions;
+  point: PathStep | null;
+  isPending?: boolean;
+};
+
+function formatChCode(chCode: string) {
+  return MAIN_OP_CH_CODES.includes(chCode) ? 'BV' : chCode;
+}
+
+const StdcmOperationalPoint = ({ updatePoint, point, isPending }: StdcmOperationalPointProps) => {
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation('stdcm');
+
+  const {
+    searchTerm,
+    chCodeFilter,
+    sortedSearchResults,
+    filteredAndSortedSearchResults,
+    setSearchTerm,
+    setChCodeFilter,
+  } = useSearchOperationalPoint({ initialSearchTerm: point?.name, initialChCodeFilter: point?.ch });
+
+  const operationalPointsSuggestions = [...sortedSearchResults].reduce(
+    (acc, p) => {
+      const newObject = {
+        label: [p.trigram, p.name].join(' '),
+        value: p.name,
+        uic: p.uic,
+      };
+      const isDuplicate = acc.some((pr) => pr.label === newObject.label);
+      if (!isDuplicate) acc.push(newObject);
+      return acc;
+    },
+    [] as { label: string; value: string; uic: number }[]
+  );
+
+  const sortedChOptions = useMemo(
+    () =>
+      [...sortedSearchResults].reduce(
+        (acc, pr) => {
+          const newObject = {
+            label: formatChCode(pr.ch),
+            id: pr.ch,
+          };
+          const isDuplicate = acc.some((option) => option.label === newObject.label);
+          if (!isDuplicate) acc.push(newObject);
+          return acc;
+        },
+        [] as { label: string; id: string }[]
+      ),
+    [point, sortedSearchResults]
+  );
+
+  const dispatchNewPoint = (p?: SearchResultItemOperationalPoint | null) => {
+    if (isNil(p)) {
+      dispatch(updatePoint(null));
+      return;
+    }
+    dispatch(
+      updatePoint({
+        name: p.name,
+        ch: p.ch,
+        id: nextId(),
+        uic: p.uic,
+        coordinates: p.geographic.coordinates,
+      })
+    );
+  };
+
+  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(e.target.value);
+    if (e.target.value.trim().length === 0) {
+      dispatchNewPoint(undefined);
+    }
+  };
+
+  const onInputOnblur = () => {
+    const newPoint =
+      operationalPointsSuggestions.length === 1
+        ? filteredAndSortedSearchResults.find(
+            (pr) => pr.name === operationalPointsSuggestions[0].value
+          )
+        : null;
```

Comment:
Use undefined here, then you can simplify the type in dispatchNewPoint (don't forget to update the test l107)
```suggestion
        : undefined;
```